src/libraries/System.Text.Json/gen/Helpers/KnownTypeSymbols.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.DotnetRuntime.Extensions;
+
+namespace System.Text.Json.Reflection
+{
+    internal sealed class KnownTypeSymbols(Compilation compilation)
```

Comment:
This is doing a ton of lookups on construction.  Are the majority of them needed on any given use?  I'm wondering if they should be lazy instead.

Reply:
> Are the majority of them needed on any given use?

Mostly, yes. There are exceptions, e.g. the attribute type symbols could be skipped in type graphs with no annotations but comparably speaking they're not many.

Reply:
FWIW this isn't changing the resolution semantics of these symbols, it simply moves them to a dedicated class. The plan is to eventually put this in an incremental value, so that each generated `JsonSerializerContext` can reference it independently and on-demand.